itlwm/hal_iwn/ItlIwn.cpp

```diff
@@ -4040,42 +4077,93 @@ iwn_set_link_quality(struct iwn_softc *sc, struct ieee80211_node *ni)
     linkq.ampdu_threshold = 3;
     linkq.ampdu_limit = htole16(4000);    /* 4ms */
 
-    i = 0;
-    if (ni->ni_flags & IEEE80211_NODE_HT) {
-        int txmcs;
-        for (txmcs = ni->ni_txmcs; txmcs >= 0; txmcs--) {
-            rinfo = &iwn_rates[iwn_mcs2ridx[txmcs]];
-            linkq.retry[i].plcp = rinfo->ht_plcp;
-            linkq.retry[i].rflags = rinfo->ht_flags;
-
-            if (ni->ni_htcaps & IEEE80211_HTCAP_SGI20)
-                linkq.retry[i].rflags |= IWN_RFLAG_SGI;
-
-            /* XXX set correct ant mask for MIMO rates here */
-            linkq.retry[i].rflags |= IWN_RFLAG_ANT(txant);
-
-            if (++i >= IWN_MAX_TX_RETRIES)
-                break;
-        }
-    } else {
-        int txrate;
-        for (txrate = ni->ni_txrate; txrate >= 0; txrate--) {
-            rinfo = &iwn_rates[wn->ridx[txrate]];
-            linkq.retry[i].plcp = rinfo->plcp;
-            linkq.retry[i].rflags = rinfo->flags;
-            linkq.retry[i].rflags |= IWN_RFLAG_ANT(txant);
-            if (++i >= IWN_MAX_TX_RETRIES)
-                break;
-        }
+    if (ic->ic_flags & IEEE80211_F_USEPROT)
+        if (sc->hw_type != IWN_HW_REV_TYPE_4965)
```

Comment:
@pigworlds  Hi :), do you remember where does the code from? It seems that it is different from Linux

Reply:
I looked at the history and can only find they were introduced in https://github.com/pigworlds/itlwm/commit/38bbc0c0dacdbe232cc49ea0d9d351b140de54b9 and https://github.com/pigworlds/itlwm/commit/bffc12ba0c846579c43c4a67043357022df86972.

I don't recall the reason for this flag, it could be a trial and error and this works.